indico/modules/events/registration/controllers/api/misc.py

```diff
@@ -5,10 +5,25 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
+from operator import itemgetter
+
+from flask import session
+
 from indico.modules.events.controllers.base import RHProtectedEventBase
+from indico.util.iterables import group_list
 
 
 class RHAPIRegistrationForms(RHProtectedEventBase):
     def _process(self):
         from indico.modules.events.registration.schemas import RegistrationFormPrincipalSchema
         return RegistrationFormPrincipalSchema(many=True).jsonify(self.event.registration_forms)
+
+
+class RHAPIEventSessionBlocks(RHProtectedEventBase):
+    def _process(self):
+        schema = [{'id': sb.id, 'start_dt': sb.start_dt.astimezone(self.event.tzinfo).date().isoformat(),
+                   'start_time': sb.start_dt.astimezone(self.event.tzinfo).strftime('%H:%M'),
+                   'end_time': sb.end_dt.astimezone(self.event.tzinfo).strftime('%H:%M'),
+                   'full_title': sb.full_title} for s in self.event.sessions for sb in s.blocks if
+                  s.can_access(session.user)]
+        return group_list(schema, key=itemgetter('start_dt'), sort_by=itemgetter('start_dt'))
```

Comment:
Much more readable IMHO:

```suggestion
        data = [
            {
                'id': sb.id,
                'start_dt': sb.start_dt.astimezone(self.event.tzinfo).date().isoformat(),
                'start_time': sb.start_dt.astimezone(self.event.tzinfo).strftime('%H:%M'),
                'end_time': sb.end_dt.astimezone(self.event.tzinfo).strftime('%H:%M'),
                'full_title': sb.full_title,
            }
            for s in self.event.sessions
            for sb in s.blocks
            if s.can_access(session.user)
        ]
        return group_list(data, key=itemgetter('start_dt'), sort_by=itemgetter('start_dt'))

```

(also changed the name to avoid calling something that's not a schema `schema`)